tensorboard/webapp/core/views/page_title_container.ts

```diff
@@ -36,12 +58,44 @@ import {State} from '../state';
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class PageTitleContainer {
-  readonly title$ = this.store.pipe(
-    select(getEnvironment),
-    map((env) => env.window_title || 'TensorBoard'),
-    // (it's an empty string when the `--window_title` flag is not set)
+  private readonly getExperimentId$ = this.store.select(getRouteKind).pipe(
+    filter((routeKind) => routeKind === RouteKind.EXPERIMENT),
+    withLatestFrom(this.store.select(getExperimentIdsFromRoute)),
+    map(([, experimentIds]) =>
+      experimentIds && experimentIds.length === 1 ? experimentIds[0] : null
+    )
+  );
+
+  private readonly experiment$ = this.getExperimentId$.pipe(
+    filter(Boolean),
```

Comment:
I haven't checked myself on a non-dashboard page (e.g. exp list, not found).  However, with this `filter(Boolean)`, I think the `experimentName$` observable would never emit, and the `title$` would never emit (I think combineLatestWith will wait for the first emission), so I think the PageTitleComponent just gets an initial `null`.

To ensure that an experiment list would get the serviceName initially, can we add a `startWith()` on the `title$` observable?

https://www.learnrxjs.io/learn-rxjs/operators/combination/startwith

Reply:
Good catch! Also changed the tests to not initialize with EXPERIMENT route to catch the scenario where the observables would not emit.